microsoft-edge/progressive-web-apps-chromium/serviceworker.md

```diff
@@ -0,0 +1,94 @@
+
+## Registering a Service Worker  
+
+Like all Web Workers, a Service Worker must exist in a separate file.  It is this file that you referenced when registering the Service Worker:  
+
+```javascript
+if ( "serviceWorker" in navigator ) {
+    navigator.serviceWorker.register( "/serviceworker.min.js" );
+}
+```  
+
+While all modern browsers support Service Workers to some extent, it is still a good idea to test for the existence of the `serviceWorker` object before running any Service Worker-related code.  Here a Service Worker is being registered using the `serviceworker.min.js` file in the root of the site.  It is critical that the JavaScript file defining a Service Worker exists in the highest-level directory you want it to govern \(which is referred to as the scope of the Service Worker\).  Here it is in the root, so it operates on every page in the domain; if this example referenced the Service Worker file within a `js` directory, the scope of the Service Worker only contains that directory and any descendants.  Unless you have good reason to reduce the scope of a Service Worker, thr best practice is to place it in the root of your site.  
+
+## The Service Worker life cycle  
+
+Every Service Worker has a life cycle, you may connect to any corresponding event generated for each step in the life cycle.  The details of the life cycle is a deep topic, the following sequence represents the Cliff's Notes version of what happens.  
+
+1.  The Service Worker is registered.  
+1.  The browser downloads the associated JavaScript and installs the Service Worker, firing the `install` event.  The `install` event is when many Service Workers pre-caches important long-lived files like the CSS, JavaScript, logo image, offline page, and so on of the website.  
+    
+    ```javascript
+    self.addEventListener( "install", function( event ){
+        console.log( "WORKER: install event in progress." );
+    });
+    ```  
+    
+1.  The Service Worker is activated, firing the `activate` event.  This is often when developers cleans up outdated caches and such.  
+    
+    ```javascript
+    self.addEventListener( "activate", event => {
+        console.log('WORKER: activate event in progress.');
+    });
+    ```  
+    
+1.  The Service Worker is ready to run, but must wait for the page to be refreshed or for the user to navigate to a new page within the site to begin working.  Developers may skip this step by using the `self.skipWaiting()` method during the `install` event.  
+    
+    ```javascript
+    self.addEventListener( "activate", event => {
+        self.skipWaiting();
+        // …
+    });
+    ```
+    
+1.  The Service Worker is running.  <!-- 🎉 -->   
```

Comment:
```suggestion
1.  The Service Worker is now running.     
```

Reply:
Can we add another step in the lifecycle for the fetch event to lead into the next section?

Reply:
Fetch is not a part of the lifecycle. Are you looking for a smoother transition?